streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -424,6 +426,7 @@
             ValueJoiner<V, V1, R> joiner,
             JoinWindows windows,
             Serde<K> keySerde,
+            Serde<V> thisValSerde,
```

Comment:
Is this only for unifying the internal implementation of different joins? Or you think it was actually a bug that we miss this serde?